web/src/containers/Main/hooks/index.ts

```diff
@@ -55,20 +132,47 @@ export function useMain(questionnaireId: string) {
         [questionnaireId, fhirMode],
     );
 
+    const [mapperInfoList, setMapperInfoList] = useState<MapperInfo[]>([]);
+    const [showModal, setShowModal] = useState(false);
+
+    useEffect(() => {
+        if (!showModal && mapperInfoList.length > 0) {
+            questionnaireManager.reload();
+            setMapperInfoList([]);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [showModal]);
+
+    const closeModal = () => setShowModal(false);
+
     const saveQuestionnaireFHIR = useCallback(
         async (resource: Questionnaire) => {
-            const response = await service({
-                method: 'PUT',
-                data: resource,
-                url: `/${fhirMode ? 'fhir/' : ''}Questionnaire/${resource.id}`,
-            });
+            const response = await updateQuestionnaire(resource, fhirMode);
             if (isSuccess(response)) {
                 questionnaireManager.reload();
+                return;
+            }
+            if (response.error.issue?.length > 0) {
+                const mappingInfoList: MapperInfo[] = [];
+                response.error.issue.map((issue, index) => {
+                    const mappingId: string | null | undefined = idExtraction(issue, resource, response);
```

Comment:
Here you already know that type of response is Failure.
You can unpack it here instead of passing as it is.
``` 
idExtraction(issue, resource, response.error);
```

As a result `idExtraction` starts accepting OperationOutcome as the third argument.